### ISSUE TITLE
GTFS-RT Should be Accessible in Shards

### ISSUE THREAD

**Issue by lolpro11** [CatenaryMaps/kactus-gtfs-rt#68]
There should be a feature to pull from a feed if you know the key value pair. Obviously an N+1 problem, but the tradeoff is worth it in certain situations

**Comment by kylerchin** [comment]
What shards?

**Comment by kylerchin** [comment]
I don't understand what this is intended to do. The GTFS-RT feed will be accessible if you know the key via a realtime database cluster. Plus, data streaming via Kafka or Remote Procedural calls will inject updates into applications that need it.

**Comment by kylerchin** [comment]
Do you mean, filtering out of the GTFS-rt set for a particular piece of information?

**Comment by lolpro11** [comment]
Yes, and eventually streaming through websocket that one entity. I need this functionality for arrival estimates and later algorithms/ML for arrival estimates.

**Comment by kylerchin** [comment]
I see. You want functionality to subscribe to a single entity?
My idea was to process arrival estimates in bulk.

**Comment by lolpro11** [comment]
Well, currently we do not have an algorithm for arrival estimates. For example, I would like to be able to filter to just one bus route in the feed in order to find the delay for that one route.

**Comment by lolpro11** [comment]
We are, at least for now, going to use swiftly/agency provided delays in the GTFS-RT, which is within a tolerance of 1 second.

**Comment by kylerchin** [comment]
Swiftly is quite good at delay calculations, I think. Transit App skips prediction for swiftly.

**Comment by lolpro11** [comment]
Route Filtering is now done on json as an example.
We will change to protobuf soon for production \;)
https://kactus.catenarymaps.org/gtfsrtasjson/?feed=f-metro~losangeles~bus~rt&category=vehicles&route=287-13172
